keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -154,37 +154,21 @@ For this reason, we can do one alpha release where we log/event if we observe us
 
 ## Design Details
 
-This can be implemented by modifying the defaults.go and validation.go components for the namespace API, i.e., in the defaults.go file for api/core:
-
-<<[UNRESOLVED See https://github.com/kubernetes/kubernetes/pull/96968 where this evolves, but this example is functionally valid, modulo warnings ]>>
-
-This will change if **warnings** are to be a core requirement of this KEP.  As that is not yet one of the goals of this KEP, however, we have a first implementation of this feature using defaults.go.
-
+This can be implemented by modifying the defaults.go and validation.go components for the namespace API, i.e., in the defaults.go file for api/core.  Note that we will support DISABLING this via a feature gate, per the versioning specification in this KEP.
 
 ```
-func SetDefaults_Namespace(obj *v1.Namespace) {
-  if obj.Labels == nil {
-  	obj.Labels = map[string]string{}
-  }
-  
-  if _, found := obj.Labels[“kubernetes.io/metadata.name”]; !found {
-  	obj.Labels[“kubernetes.io/metadata.name”] = obj.Name
-  }
-}
+		obj.Labels["kubernetes.io/metadata.name"] = obj.Name
 ```
-In validation.go, we'll go on to validate this namespace, like so:
 
+In validation.go, we'll go on to validate this namespace, like so, again, allowing disablement of the feature per the versioning specification in this KEP.
 ```
-func ValidateNamespace(namespace *core.Namespace) field.ErrorList {
-	// Check if namespace.Labels[“kubernetes.io/metadata.name”] == namespace.Name
-  // if not add an error
-  // ...
-  return allErrs
+		if namespace.Labels[v1.LabelNamespaceName] != namespace.Name {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("metadata", "labels").Key(v1.LabelNamespaceName), namespace.Labels[v1.LabelNamespaceName], fmt.Sprintf("must be %s", namespace.Name)))
+		}
 ```
-<<[/UNRESOLVED]>>
 
-This solves not just NetworkPolicy, but all such namespace selector APIs without 
-any disruption or version skew problems.
+
+This solves not just NetworkPolicy, but all such namespace selector APIs without  any disruption or version skew problems.  Note that we aren't feature gating, since this change is expected to be benign in all practical scenarios.
```

Comment:
I don't remember this discussing this claim in a larger context.  While I would happily help this feature go to alpha, bypassing our normal feature process requires broader deliberation that I don't recall having.

Reply:
@liggitt i believe mentioned in slack , going to beta on this one  today?